src/tiles/text-snippet-block.ts

```diff
@@ -0,0 +1,126 @@
+
+@customElement('text-snippet-block')
+export class TextSnippetBlock extends LitElement {
+  @property({ type: Array }) snippets?: string[] = [];
+
+  @property({ type: String }) viewSize: string = 'desktop';
+
+  render() {
+    const viewSizeClass = this.viewSize === 'grid' ? 'grid' : 'list';
+
+    if (!this.snippets || this.snippets.length === 0) return html`${nothing}`;
+
+    return html`
+      <div id="container" class=${viewSizeClass}>
+        ${this.ellipsisJoinedSnippets}
+      </div>
+
+      ${this.viewSize === 'grid' ? html`<div id="separator"></div>` : nothing}
+    `;
+  }
+
+  /**
+   * An array of HTML templates derived from the snippets, with ellipses inserted
+   * at the beginning, end, and between each pair of snippets.
+   */
+  private get ellipsisJoinedSnippets(): TemplateResult {
+    return html`&hellip; ${join(this.snippetTemplates, html` &hellip; `)}
+    &hellip;`;
+  }
+
+  /**
+   * Returns an array of HTML span templates containing this item's snippets with all of
+   * their `{{{triple-brace-delimited}}}` matches replaced by `<mark>HTML mark tags</mark>`.
+   *
+   * This approach safely avoids the use of `unsafeHTML` and leaves any existing HTML tags
+   * in the snippets intact (as inert text), rather than stripping them away with DOMPurify.
+   *
+   * Note on `<em>` vs. `<mark>`:
+   * The old search page snippets had search keywords demarcated with `<em>` tags.
+   * The `<mark>` tag is semantically more accurate for this use case than `<em>`,
+   * but screen-reader behavior may be different. `<em>` will likely be read in a
+   * different tone, while `<mark>` is often read no differently than ordinary text
+   * in many screen-readers (though there are ways to work around this if needed).
+   */
+  private get snippetTemplates(): TemplateResult[] | undefined {
+    return this.snippets?.map(s => {
+      const matches = s.matchAll(/{{{(.+?)}}}/g);
+      const templates: TemplateResult[] = [];
+
+      // Convert each match into an HTML template that includes:
+      //  - Everything from the end of the previous match (or the beginning of the
+      //      string) up to the current match, as raw text.
+      //  - The current match (excluding the curly braces) wrapped in a `<mark>` tag.
+      let index = 0;
+      for (const match of matches) {
+        if (match.index != null) {
+          templates.push(html`
+            ${s.slice(index, match.index)}
+            <mark>${match[1]}</mark>
+          `);
+          index = match.index + match[0].length;
+        }
+      }
+
+      // Include any text from the last match to the end
+      templates.push(html`${s.slice(index)}`);
+
+      // Squash everything into a single span template
+      return html`<span>${templates}</span>`;
+    });
+  }
+
+  static get styles(): CSSResultGroup {
+    return css`
+      #container {
+        display: -webkit-box;
+        font-family: 'Times New Roman', serif;
+        color: #2c2c2c;
```

Comment:
you _probably_ don't need `color` as it is a global css var that cascades into Shadow DOM...
Q: does this text color always match the tile's main color? if so, you may not need to set ^_^ good to double check.

Reply:
Yep, looks like this is the same as the inherited value so not needed! 👍